idunn/api/search.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from fastapi import Depends
+
+from idunn import settings
+from idunn.geocoder.bragi_client import bragi_client
+from idunn.geocoder.nlu_client import nlu_client, NluClientException
+from idunn.utils import result_filter
+from idunn.instant_answer import normalize
+from ..geocoder.models import QueryParams, IdunnAutocomplete
+
+logger = logging.getLogger(__name__)
+
+nlu_allowed_languages = settings["NLU_ALLOWED_LANGUAGES"].split(",")
+
+
+def no_search_result(query=None, lang=None):
+    if query is not None:
+        logger.info(
+            "search: no result",
+            extra={
+                "request": {
+                    "query": query,
+                    "lang": lang,
+                }
+            },
+        )
+    return IdunnAutocomplete()
+
+
+async def search(query: QueryParams = Depends(QueryParams)) -> IdunnAutocomplete:
```

Comment:
Only `q` and `lang` query are supported. So the documentation would be misleading for this route.  For example `?nlu=false` would not have any effect.

Is there any obstacle to handle the same query parameters as `/v1/autocomplete` ?

Reply:
I agree with you that it sounds nicer to support the same set of parameters as `/v1/autocomplete`, I had some issues with it on a previous implementation but there is nothing holding it back now, this is fixed!